#!/usr/bin/env python
PACKAGE = "perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("bottom_left_loc_x", int_t, 0,   "bottom left location (x-axis) of the ego vehicle relative to recorded image size. ", 270, 0, 1280)
gen.add("bottom_left_loc_y", int_t, 0,  "bottom left location (y-axis) of the ego vehicle relative to recorded image size. ", 480, 0, 1024)
gen.add("top_left_loc_x", int_t, 0,      "top left location (x-axis) of the ego vehicle relative to recorded image size. ", 270, 0, 1280)
gen.add("top_left_loc_y", int_t, 0,      "top left location (y-axis) of the ego vehicle relative to recorded image size.", 360, 0, 1024)
gen.add("top_right_loc_x", int_t, 0,     "top right location (x-axis) of the ego vehicle relative to recorded image size.", 500, 0, 1280)
gen.add("top_right_loc_y", int_t, 0,     "top right location (y-axis) of the ego vehicle relative to recorded image size.", 360, 0, 1024)
gen.add("bottom_right_loc_x", int_t, 0,  "bottom right location (x-axis) of the ego vehicle relative to recorded image size.", 500, 0, 1280)
gen.add("bottom_right_loc_y", int_t, 0,  "bottom right location (y-axis) of the ego vehicle relative to recorded image size.", 480, 0, 1024)

exit(gen.generate(PACKAGE, "perception", "PreprocessingTrapezoidMask"))
